spyder/plugins/completion/languageserver/plugin.py

```diff
@@ -261,8 +268,12 @@ def project_path_update(self, project_path, update_kind):
     @Slot(str)
     def report_server_error(self, error):
         """Report server errors in our error report dialog."""
-        self.main.console.exception_occurred(error, is_traceback=True,
-                                             is_pyls_error=True)
+        error_data = dict(
+            text=error,
+            is_traceback=True,
+            is_pyls_error=True,
+        )
```

Comment:
There is no need to be fixed for this PR, but can you think of a way this can be generalized?

I mean that the internal console that handles these exceptions had to define a method just for this by having the

`is_pyls_error=True`, that feels like a code smell.

Reply:
I mean that the console should probably only care about the text and it being a traceback or not. If we need to do some special processing of that error, it should be done on the side of this plugin and not the console.

Reply:
We talked internally about how this could be accomplished, so using the `dict` but have for now "error", "is_traceback", a "title"  option and a "repo"  one so that any plugin (external or internal) can call upon this dialog and provide enough information so that the error reporter on the console remains as generic and ignorant from whatever plugin called. When in doubt asking oneself how an external plugin could do this, usually helps a lot in understanding how we are special casing plugins and tying them together.

To find internal plugins you can use https://github.com/spyder-ide/spyder/blob/master/spyder/app/solver.py#L27 @ccordoba12

Reply:
Fixed